# SQL Analysis Report

This report summarizes the main business insights obtained through SQL analysis of the Italian Restaurant Analytics project.

The analysis covers sales, promotions, payment methods, employees, and products using data stored in a SQLite database.

In [1]:
# Import the libraries required to connect to SQLite and display query results

import pandas as pd
import sqlite3

## Database Connection

The report connects to the SQLite database created from the cleaned restaurant datasets.

In [2]:
# Connect to the SQLite database

conn = sqlite3.connect("../sql/restaurant_analytics.db")

print("Database connection created successfully.")

Database connection created successfully.


## Sales Analysis

This section summarizes the main sales results obtained through SQL queries, including total sales, sales by category, and monthly performance.


### Total Sales

The following SQL query calculates total sales during the analyzed period.

In [4]:
# Execute the SQL query for total sales

query = """
SELECT
    ROUND(SUM("Sales"), 2) AS Total_Sales
FROM sales;
"""

result = pd.read_sql_query(query, conn)

result

,Total_Sales
0,1344807.28


### Key Insight

Total sales during the analyzed period reached $1,344,807.28.

### Total Sales by Category

The following SQL query calculates total sales for each business category.

In [3]:
# Execute the SQL query for total sales by category

query = """
SELECT
    "Category",
    ROUND(SUM("Sales"), 2) AS Total_Sales
FROM sales
GROUP BY "Category"
ORDER BY Total_Sales DESC;
"""

result = pd.read_sql_query(query, conn)

result

,Category,Total_Sales
0,Food,948232.21
1,Liquor,211747.70
2,Beverages,157079.85
3,Events,27747.52


### Key Insight

Food was the dominant sales category during the analyzed period, generating $948,232.21 in total sales and representing approximately 70.51% of overall revenue.

### Monthly Sales

The following SQL query summarizes total sales by month.

In [5]:
# Execute the SQL query for monthly sales

query = """
SELECT
    strftime('%Y-%m', "Date") AS Month,
    ROUND(SUM("Sales"), 2) AS Total_Sales
FROM sales
GROUP BY Month
ORDER BY Month;
"""

result = pd.read_sql_query(query, conn)

result

,Month,Total_Sales
0,2025-03,103876.35
1,2025-04,86374.69
2,2025-05,102869.21
3,2025-06,100154.40
4,2025-07,100600.17
5,2025-08,98279.11
6,2025-09,91959.74
7,2025-10,102360.35
8,2025-11,114970.30
9,2025-12,131135.53


### Key Insight

Monthly sales varied throughout the analyzed period, with December 2025 recording the highest total sales.

### Highest-Sales Month

The following SQL query identifies the month with the highest total sales.

In [6]:
# Execute the SQL query for the highest-sales month

query = """
SELECT
    strftime('%Y-%m', "Date") AS Month,
    ROUND(SUM("Sales"), 2) AS Total_Sales
FROM sales
GROUP BY Month
ORDER BY Total_Sales DESC
LIMIT 1;
"""

result = pd.read_sql_query(query, conn)

result

,Month,Total_Sales
0,2025-12,131135.53


### Key Insight

December 2025 was the strongest month, generating $131,135.53 in total sales.

## Promotions Analysis

This section summarizes promotion usage and discount amounts during the analyzed period.

### Total Discount Amount

The following SQL query calculates the total discount amount generated by promotions.

In [7]:
# Execute the SQL query for total discount amount

query = """
SELECT
    ROUND(SUM("Discount Amount"), 2) AS Total_Discount_Amount
FROM promotions;
"""

result = pd.read_sql_query(query, conn)

result

,Total_Discount_Amount
0,90189.53


### Key Insight

Promotions generated a total discount amount of $90,189.53 during the analyzed period.

### Most Used Promotions

The following SQL query identifies the ten promotions with the highest number of uses.

In [8]:
# Execute the SQL query for the most used promotions

query = """
SELECT
    "Promotion",
    SUM("Uses") AS Total_Uses
FROM promotions
GROUP BY "Promotion"
ORDER BY Total_Uses DESC
LIMIT 10;
"""

result = pd.read_sql_query(query, conn)

result

,Promotion,Total_Uses
0,3x2 SIEMPRE,3823
1,10% GRANDE TABLE,2438
2,100% PROPIETARIOS,541
3,CORTESIA AL PRODUCTO,351
4,15% DESC. EMPLEADOS,218
5,V&E&L FEBRERO 2026,159
6,50% MODO TASTY,141
7,15 % DESC. WHISKYS,138
8,10% DESCUENTO BARRA,121
9,DESC MODO TASTY 50%,88


### Key Insight

`3x2 SIEMPRE` was the most frequently used promotion with 3,823 uses, followed by `10% GRANDE TABLE` with 2,438 uses.

### Highest Discount Amount by Promotion

The following SQL query identifies the promotions that generated the highest total discount amounts.

In [9]:
# Execute the SQL query for promotions with the highest discount amounts

query = """
SELECT
    "Promotion",
    ROUND(SUM("Discount Amount"), 2) AS Total_Discount_Amount
FROM promotions
GROUP BY "Promotion"
ORDER BY Total_Discount_Amount DESC
LIMIT 10;
"""

result = pd.read_sql_query(query, conn)

result

,Promotion,Total_Discount_Amount
0,3x2 SIEMPRE,45045.70
1,10% GRANDE TABLE,21141.53
2,100% PROPIETARIOS,10052.87
3,50% MODO TASTY,4945.45
4,CORTESIA AL PRODUCTO,2072.96
5,15 % DESC. WHISKYS,1402.61
6,V&E&L FEBRERO 2026,1171.92
7,DESC MODO TASTY 50%,974.02
8,30% PROPIETARIOS,766.34
9,15% DESC. BOCATTO,619.61


### Key Insight

`3x2 SIEMPRE` generated the highest total discount amount at $45,045.70, followed by `10% GRANDE TABLE` at $21,141.53.

## Payment Methods Analysis

This section summarizes sales and transaction activity across the different payment methods used during the analyzed period.

### Sales by Payment Method

The following SQL query calculates total sales generated by each payment method.

In [11]:
# Execute the SQL query for sales by payment method

query = """
SELECT
    "Payment Method",
    ROUND(SUM("Sales"), 2) AS Total_Sales
FROM payment_methods
GROUP BY "Payment Method"
ORDER BY Total_Sales DESC;
"""

result = pd.read_sql_query(query, conn)

result

,Payment Method,Total_Sales
0,TC. DINERS,395902.18
1,TC. VISA,353691.02
2,TD. VISA,274220.15
3,TC. MASTERCARD,257354.16
4,EFECTIVO,92799.98
5,TC.AMEX,44562.36
6,UBEREATS,39221.83
7,TC. DISCOVER,21414.99
8,RAPPI,19264.04
9,CUENTA X COBRAR,18002.04


### Key Insight

`TC. DINERS` generated the highest total sales at $395,902.18, followed by `TC. VISA` with $353,691.02 and `TD. VISA` with $274,220.15.

### Transactions by Payment Method

The following SQL query calculates the total number of transactions for each payment method.

In [12]:
# Execute the SQL query for transactions by payment method

query = """
SELECT
    "Payment Method",
    SUM("Transactions") AS Total_Transactions
FROM payment_methods
GROUP BY "Payment Method"
ORDER BY Total_Transactions DESC;
"""

result = pd.read_sql_query(query, conn)

result

,Payment Method,Total_Transactions
0,TC. DINERS,6386
1,TC. VISA,5501
2,TD. VISA,4848
3,TC. MASTERCARD,4037
4,EFECTIVO,2565
5,UBEREATS,1315
6,RAPPI,784
7,TC.AMEX,638
8,MERMAS,617
9,MANUAL,595


### Key Insight

`TC. DINERS` recorded the highest number of transactions with 6,386, followed by `TC. VISA` with 5,501 and `TD. VISA` with 4,848 transactions.

### Monthly Sales by Payment Method

The following SQL query summarizes total sales by payment method for each month.

In [13]:
# Execute the SQL query for monthly sales by payment method

query = """
SELECT
    strftime('%Y-%m', "Date") AS Month,
    "Payment Method",
    ROUND(SUM("Sales"), 2) AS Total_Sales
FROM payment_methods
GROUP BY Month, "Payment Method"
ORDER BY Month, Total_Sales DESC;
"""

result = pd.read_sql_query(query, conn)

result

,Month,Payment Method,Total_Sales
0,2025-03,TC. DINERS,34480.09
1,2025-03,TC. VISA,24549.31
2,2025-03,TD. VISA,20143.59
3,2025-03,TC. MASTERCARD,19137.92
4,2025-03,EFECTIVO,7957.73
...,...,...,...
222,2026-03,CONSUMO INTERNO,288.33
223,2026-03,FACT REPROCESADA,189.99
224,2026-03,ORDEN DE CONSUMO,160.19
225,2026-03,TRANSFERENCIA,146.79


### Key Insight

Payment method performance varied across the analyzed months, with card payments consistently representing the strongest sales contribution.

## Employees Analysis

This section summarizes employee performance based on total sales and quantity sold during the analyzed period.

### Employee Sales Ranking

The following SQL query ranks employees by total sales.

In [14]:
# Execute the SQL query for employee sales ranking

query = """
SELECT
    "Employee",
    ROUND("Sales", 2) AS Total_Sales
FROM employees
ORDER BY Total_Sales DESC;
"""

result = pd.read_sql_query(query, conn)

result

,Employee,Total_Sales
0,Employee 1,61641.63
1,Employee 2,54125.31
2,Employee 3,45762.89
3,Employee 4,42810.66
4,Employee 5,40654.99
5,Employee 6,24884.01
6,Employee 7,20890.97
7,Employee 8,8458.35
8,Employee 9,5343.32
9,Employee 10,4883.78


### Key Insight

`Employee 1` generated the highest total sales at $61,641.63, followed by `Employee 2` with $54,125.31 and `Employee 3` with $45,762.89.

### Employee Quantity Ranking

The following SQL query ranks employees by total quantity sold.

In [15]:
# Execute the SQL query for employee quantity ranking

query = """
SELECT
    "Employee",
    ROUND("Quantity", 2) AS Total_Quantity
FROM employees
ORDER BY Total_Quantity DESC;
"""

result = pd.read_sql_query(query, conn)

result

,Employee,Total_Quantity
0,Employee 1,15786.20
1,Employee 2,14256.90
2,Employee 3,12638.87
3,Employee 4,11382.50
4,Employee 5,10697.50
5,Employee 7,5788.33
6,Employee 6,3617.62
7,Employee 8,2264.91
8,Employee 9,1394.33
9,Employee 10,1305.00


### Key Insight

`Employee 1` recorded the highest total quantity sold with 15,786.20 units, followed by `Employee 2` with 14,256.90 and `Employee 3` with 12,638.87 units.

## Products Analysis

This section summarizes product performance based on sales, quantity sold, category contribution, and average sales value per unit.

### Top Products by Sales

The following SQL query identifies the fifteen products with the highest total sales.

In [16]:
# Execute the SQL query for top products by sales

query = """
SELECT
    "Product",
    ROUND("Sales", 2) AS Total_Sales
FROM products
ORDER BY Total_Sales DESC
LIMIT 15;
"""

result = pd.read_sql_query(query, conn)

result

,Product,Total_Sales
0,Lomo Fino A La Parrilla,9538.15
1,Pasta Favorita Del Chef,7966.39
2,Salmone,7405.87
3,Chicken Parmegiana,6467.60
4,Lasagna,6465.16
5,Carbonara,6334.68
6,Aperol Spritz,6223.70
7,Risotto Al Lomo Saltado,6093.58
8,Picana Angus Importada,6076.00
9,Pulpo a la Parrilla,6073.65


### Key Insight

`Lomo Fino A La Parrilla` was the highest-selling product, generating $9,538.15 in total sales, followed by `Pasta Favorita Del Chef` with $7,966.39 and `Salmone` with $7,405.87.

### Top Products by Quantity Sold

The following SQL query identifies the fifteen products with the highest quantity sold.

In [17]:
# Execute the SQL query for top products by quantity sold

query = """
SELECT
    "Product",
    ROUND("Quantity", 2) AS Total_Quantity
FROM products
ORDER BY Total_Quantity DESC
LIMIT 15;
"""

result = pd.read_sql_query(query, conn)

result

,Product,Total_Quantity
0,3 X 2 Siempre,1930.0
1,Agua Sin Gas San Felipe,1734.0
2,Limonada Frutos Rojos,1464.0
3,Guitig Clasica,1244.0
4,Guitig,902.0
5,Limonada De Mandarina,811.0
6,Empaque Delivery,811.0
7,Coca Zero 300CC,809.0
8,Limonada Con H.B,793.0
9,Limonada Imperial,775.0


### Key Insight

`3 X 2 Siempre` recorded the highest quantity with 1,930 units, followed by `Agua Sin Gas San Felipe` with 1,734 units and `Limonada Frutos Rojos` with 1,464 units.

Several of the highest-volume products were beverages, promotional items, or operational items, showing that high quantity does not necessarily correspond to high revenue.

### Sales by Product Category

The following SQL query calculates total sales for each product category.

In [18]:
# Execute the SQL query for sales by product category

query = """
SELECT
    "Category",
    ROUND(SUM("Sales"), 2) AS Total_Sales
FROM products
GROUP BY "Category"
ORDER BY Total_Sales DESC;
"""

result = pd.read_sql_query(query, conn)

result

,Category,Total_Sales
0,Pasta,51609.07
1,"Meat, chicken & seafood",45586.08
2,3x2 Promotion,37675.42
3,Entrees,25276.99
4,Specials,23341.59
5,Juice,19103.62
6,Pizza & Calzone,18354.51
7,Risotto,13655.13
8,Salads,10107.59
9,Water,7667.81


### Key Insight

`Pasta` generated the highest total sales at $51,609.07, followed by `Meat, chicken & seafood` with $45,586.08 and `3x2 Promotion` with $37,675.42.

### Average Sales Value per Unit

The following SQL query identifies products with the highest average sales value per unit among products with at least 20 units sold.

In [19]:
# Execute the SQL query for average sales value per unit

query = """
SELECT
    "Product",
    ROUND("Sales" / "Quantity", 2) AS Average_Sales_Per_Unit
FROM products
WHERE "Quantity" >= 20
ORDER BY Average_Sales_Per_Unit DESC
LIMIT 15;
"""

result = pd.read_sql_query(query, conn)

result

,Product,Average_Sales_Per_Unit
0,Rack De Cordero Importado,36.68
1,Bife De Chorizo,34.92
2,Picana Angus Importada,27.74
3,Risotto Frutti Di Mare,25.50
4,Asap,24.90
5,Pasta Frutti Di Mare,24.05
6,Pesca A La Parrilla,23.72
7,Risotto Al Lomo Saltado,23.26
8,Lomo Saltado Huancaina,23.20
9,Salmone,23.14


### Key Insight

Among products with at least 20 units sold, `Rack De Cordero Importado` recorded the highest average sales value per unit at $36.68, followed by `Bife De Chorizo` at $34.92 and `Picana Angus Importada` at $27.74.

Average sales value per unit should not be interpreted as profit margin, since product costs are not included in the dataset.

## Final Conclusions

The SQL analysis confirmed the main business patterns identified across the restaurant datasets.

Sales were strongly concentrated in the Food category, while December 2025 recorded the highest monthly sales during the analyzed period.

Promotional activity was dominated by `3x2 SIEMPRE`, both in usage and total discount amount.

Card payment methods generated the highest sales and transaction volumes, with `TC. DINERS` leading both metrics.

Employee performance showed a clear concentration among the top performers, with `Employee 1` ranking first in both sales and quantity sold.

Product analysis showed that the highest-selling products were not always the highest-volume products. Premium dishes generated higher sales value per unit, while several beverages and promotional items recorded high quantities with comparatively lower revenue.

Overall, SQL provided a structured and reproducible way to summarize, rank, filter, and compare the main business metrics across the restaurant datasets.

---

## Author

**Guido Acosta**

Data Analyst Portfolio Project

GitHub Portfolio – 2026